### PR TITLE
[cxx-interop] Fix friend operators that come from class template specializations.

### PR DESCRIPTION
**Pull request by zoecarver**

### CODE REVIEW

File: test/Interop/Cxx/stdlib/use-std-map.swift [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+// REQUIRES: OS=macosx
+
+import StdlibUnittest
+import StdMap
+import std.map
+import Cxx
+
+var StdMapTestSuite = TestSuite("std::map")
+
+extension Map.const_iterator : UnsafeCxxInputIterator { }
```
Comment (zoecarver) [inline test/Interop/Cxx/stdlib/use-std-map.swift:15]: https://github.com/llvm/llvm-project/blob/main/libcxx/include/map#L954-L959

**Review by ktoso — COMMENTED** [review]
Awesome, thank you 👍

**Comment by ktoso** [comment]
This is kind of related to: rdar://100021897 as it'd mean we don't have to use `[]` but just could use the `find`

**Comment by zoecarver** [comment]
@swift-ci please test

File: test/Interop/Cxx/stdlib/use-std-map.swift [hunk 2]
```diff
@@ -7,6 +7,8 @@
 
 import StdlibUnittest
 import StdMap
+import std.map
```
Comment (zoecarver) [inline test/Interop/Cxx/stdlib/use-std-map.swift:10]: @egorzhdan am I supposed to just use `StdMap`. If so, that caused problems locally (looking up operator==, I think).

**Comment by zoecarver** [comment]
@swift-ci please smoke test

**Pull request merged**